ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -937,9 +943,9 @@ protected ByteBuffer filterChunkToRange(ByteBuffer buf) {
         buf.position(0);
         buf.limit(0);
       } else {
-        long startOffsetInThisChunk = chunkIndex == 0 ? resolvedByteRange.getStartOffset() % chunkSize : 0;
```

Comment:
I wonder previously if `chunkIndex > 0`, why the `startOffsetInThisChunk` is 0 here?

Reply:
because it's relative to the chunk itself. The chunkIndexes only refer to chunks that are within a byte range.  If the chunkIndex refers to a chunk beyond the first chunk, it will have to start at 0.

Reply:
Sorry about the confusion around `chunkIndex`. Since we now have the `offset` variable, you could consider changing this to `long startOffsetInThisChunk = Math.max(resolvedByteRange.getStartOffset() - offset, 0)` if you prefer.